infrastructure/zk/src/format_sql.ts

```diff
@@ -0,0 +1,161 @@
+        let currentLine = '';
+        currentLine += i == 0 ? '"' : ' ';
+        currentLine += formattedLines[i];
+        currentLine += i != formattedLines.length - 1 ? ' \\' : '';
+        currentLine += i == formattedLines.length - 1 ? '"' : '';
+
+        formattedLines[i] = currentLine;
+    }
+    return formattedLines.join('\n');
+}
+
+function addIndent(query: string, indent: number) {
+    return query
+        .split('\n')
+        .map((line) => ' '.repeat(indent) + line)
+        .join('\n');
+}
+
+function formatRustStringQuery(query: string) {
+    const baseIndent = query.search(/\S/);
+    let endedWithAComma = query.trim().endsWith(',');
+
+    const rawQuery = extractQueryFromRustString(query);
+    const formattedQuery = formatQuery(rawQuery);
+    const reconstructedRustString = embedTextInsideRustString(formattedQuery);
+
+    return addIndent(reconstructedRustString, baseIndent) + (endedWithAComma ? ',' : '') + '\n';
+}
+
+async function formatFile(filePath: string, check: boolean) {
+    let content = await fs.promises.readFile(filePath, { encoding: 'utf-8' });
+    let linesToQuery = null;
+    let isInsideQuery = false;
+    let isRawString = false;
+    let builtQuery = '';
+
+    let modifiedFile = '';
+    for (const line of content.split('\n')) {
+        if (line.endsWith('sqlx::query!(')) {
+            linesToQuery = 1;
+            isRawString = false;
+            builtQuery = '';
+        }
+        if (line.endsWith('sqlx::query_as!(')) {
+            linesToQuery = 2;
+            isRawString = false;
+            builtQuery = '';
+        }
+
+        if (linesToQuery !== null) {
+            if (linesToQuery == 0) {
+                isInsideQuery = true;
+                linesToQuery = null;
+                if (line.includes('r#"')) {
+                    isRawString = true;
+                }
+            } else {
+                linesToQuery -= 1;
+            }
+        }
+
+        if (isInsideQuery) {
+            const rawStringQueryEnded = (line.endsWith('"#,') || line.endsWith('"#')) && builtQuery;
+            const regularStringQueryEnded = (line.endsWith('",') || line.endsWith('"')) && builtQuery;
+            builtQuery += line + '\n';
+            const lineEndIsNotEscape = !line.endsWith('\\"') && !line.endsWith('\\",');
+            if (
+                (isRawString && rawStringQueryEnded) ||
+                (!isRawString && regularStringQueryEnded && lineEndIsNotEscape)
+            ) {
+                isInsideQuery = false;
+                let formattedQuery = formatRustStringQuery(builtQuery);
+                modifiedFile += formattedQuery;
+            }
+        } else {
+            modifiedFile += line + '\n';
+        }
+    }
+    modifiedFile = modifiedFile.slice(0, modifiedFile.length - 1);
+
+    if (!check) {
+        await fs.promises.writeFile(filePath, modifiedFile);
+    } else {
+        if (content !== modifiedFile) {
```

Comment:
Nit: Can't this be collapsed with the preceding `else` into `else if`?

Reply:
@tomg10 I agree it's not a big deal but I'm a bit suprized - sounds like an almost perfect usecase for `else if`